build/transpile.js

```diff
@@ -870,6 +896,17 @@ class Transpiler {
             php.push (phpBody);
             php.push ('    }')
 
+            // compile signature + body for PHP Async using generators
+            phpAsync.push('');
+            if (keyword) {
+                phpAsync.push('    public function ' + method + '(' + phpArgs + ') : Generator {');
```

Comment:
i dont think this will work in php5, which ideally we would like to support

Reply:
Correct, it won't, I've noted in the docs too. I gotta say, support for PHP5 was a great idea a few years ago, but I think increasingly not important. But let's not open that debate! 

Users with php5 can still use the library, just not the new ccxt_async namespace (ie the asynchronous syntax/functions), ie will still work for them just fine using the current approach.

Reply:
Specifically, coroutines are only php5.5 feature, and recoilphp claims only compatible with php7 (not sure why). So there's no way to get this syntax I use working with php5.4

Reply:
yeah we already have some code in ccxt that doesn't work in php5, namely keccak signatures and some random `print` function